### ISSUE TITLE
Add aria labels to responsive columns

### ISSUE THREAD

**Issue by huwshimi** [canonical-web-and-design/react-components#701]
When using a responsive variant of the MainTable you need to add an aria-label to each column for the responsive descriptions to be displayed correctly, e.g. `{ "aria-label": "VLAN", content: <VLANLink id={vlan.id} /> }`

_Originally posted by @petermakowski in https://github.com/canonical-web-and-design/maas-ui/pull/3582#discussion_r798665302_

**Comment by bartaz** [comment]
Since Vanilla 3.0 it should be `data-heading` instead of `aria-label`, because `aria-label` should not be used in this context as it makes screen reader read only the label and not contents of the cell.
